lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts

```diff
@@ -135,32 +141,47 @@ export class TaskListCloudComponent extends BaseTaskListCloudComponent {
     @Input()
     candidateGroupId: string = '';
 
+    rows: TaskInstanceCloudListViewModel[] = [];
+    dataAdapter: TasksListDatatableAdapter | undefined;
+
     constructor(@Inject(TASK_LIST_CLOUD_TOKEN) public taskListCloudService: TaskListCloudServiceInterface,
                 appConfigService: AppConfigService,
                 taskCloudService: TaskCloudService,
                 userPreferences: UserPreferencesService,
-                @Inject(TASK_LIST_PREFERENCES_SERVICE_TOKEN) cloudPreferenceService: PreferenceCloudServiceInterface) {
+                @Inject(TASK_LIST_PREFERENCES_SERVICE_TOKEN) cloudPreferenceService: PreferenceCloudServiceInterface,
+                private viewModelCreator: VariableMapperService
+            ) {
         super(appConfigService, taskCloudService, userPreferences, PRESET_KEY, cloudPreferenceService);
     }
 
-    load(requestNode: TaskQueryCloudRequestModel) {
+    loadTasks() {
         this.isLoading = true;
 
-        combineLatest([
-            this.taskListCloudService.getTaskByRequest(requestNode),
-            this.isColumnSchemaCreated$
-        ]).pipe(
-            take(1)
-        ).subscribe(
-            ([tasks]) => {
-                this.rows = tasks.list.entries;
-                this.success.emit(tasks);
-                this.isLoading = false;
-                this.pagination.next(tasks.list.pagination);
-            }, (error) => {
-                this.error.emit(error);
-                this.isLoading = false;
-            });
+        this.isColumnSchemaCreated$.pipe(
+            take(1),
+            switchMap(() => of(this.createRequestNode())),
+            tap((requestNode) => this.requestNode = requestNode),
+            switchMap((requestNode) => this.taskListCloudService.getTaskByRequest(requestNode))
+        ).subscribe((tasks: { list: PaginatedEntries<TaskCloudModel> }) => {
+            const tasksWithVariables = tasks.list.entries.map((task) => ({
```

Comment:
here I only preparing it for `mapVariablesByColumnTitle`, which require the `variable` key. For processes we are receiving `variables` for tasks `processVariables`, I think it would be redundant :)

It would be something like:

```
const tasksWithVariables: (TaskCloudModel & { variables?: ProcessInstanceVariable[] })[] = tasks.list.entries.map((task) => ({
```

Reply:
ok thanks